sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -263,6 +264,51 @@ case class PreprocessTableCreation(sparkSession: SparkSession) extends Rule[Logi
 
         create.withPartitioning(normalizedPartitions)
       }
+
+    case create: V2CreateTablePlanX if create.childrenResolved =>
```

Comment:
The original `case create: V2CreateTablePlan` will be removed and this will be changed to
`case create: V2CreateTablePlan` after the migration of `CreateTableAsSelect`, `ReplaceTable` and `ReplaceTableAsSelect` is done.